packages/openapi3/src/openapi.ts

```diff
@@ -270,7 +276,14 @@ function createOAPIEmitter(program: Program, options: OpenAPIEmitterOptions) {
     currentBasePath = basePathForRoute(program, route);
 
     for (const [_, op] of route.operations) {
-      emitEndpoint(route, op);
+      if (currentVersion) {
+        const projectedOp = program.checker!.project(op, op.projections[0].to!, [
```

Comment:
We should have a pattern here for getting a well known projection rather than assuming [0]